pkg/transient-disk/transient-disk.go

```diff
@@ -0,0 +1,127 @@
+)
+
+func generateVMBaseDir(vm *v1.VirtualMachine) string {
+	return fmt.Sprintf("%s", mountBaseDir)
+}
+func generateVolumeMountDir(vm *v1.VirtualMachine) string {
+	baseDir := generateVMBaseDir(vm)
+	return fmt.Sprintf("%s", baseDir)
+}
+
+func GetFilePath(vm *v1.VirtualMachine, volumeName string) (string, string, error) {
+	volumeMountDir := generateVolumeMountDir(vm)
+	return volumeMountDir + "/" + volumeName + ".qcow2", "qcow2", nil
+}
+
+func GenerateVolumes(vm *v1.VirtualMachine) ([]kubev1.Volume, error) {
+	var volumes []kubev1.Volume
+
+	for _, volume := range vm.Spec.Volumes {
+		if volume.Transient != nil {
+			volumes = append(volumes, kubev1.Volume{
+				Name: "transientdisk",
+				VolumeSource: kubev1.VolumeSource{
+					EmptyDir: &kubev1.EmptyDirVolumeSource{},
+				},
+			})
+
+			// We use one EmptyDir per multiple transientdisks, therefore only the first occurence matters.
+			return volumes, nil
+		}
+	}
+
+	return volumes, nil
+}
+
+func GenerateContainers(vm *v1.VirtualMachine, podVolumeName string, podVolumeMountDir string) ([]kubev1.Container, error) {
+	var containers []kubev1.Container
+
+	initialDelaySeconds := 2
+	timeoutSeconds := 5
+	periodSeconds := 5
+	successThreshold := 2
+	failureThreshold := 5
+
+	// Make VM Image Wrapper Containers
+	for _, volume := range vm.Spec.Volumes {
+		if volume.Transient != nil {
+			diskContainerName := fmt.Sprintf("volume%s", volume.Name)
+			diskContainerImage := transientContainerImage
+			filePath, _, _ := GetFilePath(vm, volume.Name)
+
+			baseImageFile, _, err := getFilePathByVolume(vm, volume)
+			if err != nil {
+				return containers, err
+			}
+
+			containers = append(containers, kubev1.Container{
+				Name:            diskContainerName,
+				Image:           diskContainerImage,
+				ImagePullPolicy: kubev1.PullIfNotPresent,
+				Command:         []string{"/entry-point.sh"},
+				Env: []kubev1.EnvVar{
+					{
+						Name:  "BASE_IMAGE",
+						Value: baseImageFile,
+					},
+					{
+						Name:  "COW_IMAGE",
+						Value: filePath,
+					},
+				},
+				VolumeMounts: []kubev1.VolumeMount{
+					{
+						Name:      podVolumeName,
+						MountPath: podVolumeMountDir,
+					},
+				},
+				// The readiness probes ensure the volume coversion and copy finished
+				// before the container is marked as "Ready: True"
+				ReadinessProbe: &kubev1.Probe{
+					Handler: kubev1.Handler{
+						Exec: &kubev1.ExecAction{
+							Command: []string{
+								"cat",
+								"/tmp/healthy",
+							},
+						},
+					},
+					InitialDelaySeconds: int32(initialDelaySeconds),
+					PeriodSeconds:       int32(periodSeconds),
+					TimeoutSeconds:      int32(timeoutSeconds),
+					SuccessThreshold:    int32(successThreshold),
+					FailureThreshold:    int32(failureThreshold),
+				},
+			})
+		}
+	}
+
+	return containers, nil
+}
+
+func getFilePathByVolume(vm *v1.VirtualMachine, src v1.Volume) (string, string, error) {
+	if src.RegistryDisk == nil {
```

Comment:
this line looks like it prevents transient disks from working. I think you want to change this to look for the PVC file now.